src/esp/bindings/SensorBindings.cpp

```diff
@@ -37,6 +38,94 @@ esp::scene::SceneNode* nodeGetter(T& self) {
 namespace esp {
 namespace sensor {
 
+// Get the torch tensor or numpy array buffer of the Sensor, initialize if it
+// does not exist yet
+template <class T>
+
+auto buffer(T& self, int gpuDevice) {
+  py::handle handle = py::cast(self);
+  if (!py::hasattr(handle, "__buffer")) {
+    if (self.specification()->gpu2gpuTransfer) {
+#ifdef ESP_BUILD_WITH_CUDA
+      auto torch = py::module_::import("torch");
+      if (self.specification()->sensorType == SensorType::Semantic) {
+          py::setattr(
+              handle, "__buffer",
+              torch.attr("empty")(
+                  (py::int_(self.specification()->resolution[0]),
+                    py::int_(self.specification()->resolution[1])),
+                  "dtype"_a = torch.attr("int32"),
+                  "device"_a = torch.attr("device")(
+                      py::str("cuda"), py::int_(gpuDevice)));
+      } else if (self.specification()->sensorType == SensorType::Depth) {
+        py::setattr(
+            handle, "__buffer",
+            torch.attr("empty")((py::int_(self.specification()->resolution[0]),
+                                 py::int_(self.specification()->resolution[1])),
+                                "dtype"_a = torch.attr("float32"),
+                                "device"_a = torch.attr("device")(
+                                    py::str("cuda"), py::int_(gpuDevice))));
+      } else {
+        py::setattr(
+            handle, "__buffer",
+            torch.attr("empty")((py::int_(self.specification()->resolution[0]),
+                                 py::int_(self.specification()->resolution[1]),
+                                 py::int_(self.specification()->channels)),
+                                "dtype"_a = torch.attr("uint32"),
+                                "device"_a = torch.attr("device")(
+                                    py::str("cuda"), py::int_(gpuDevice))));
+      }
+#endif
+    } else {
+      if (self.specification()->sensorType == SensorType::Semantic) {
+        auto pyBuffer = py::array(py::buffer_info(
```

Comment:
This part isn't even necessary, you can return the array plainly by reference. @mosra Thoughts? Also not sure setting the buffer here like this is the appropriate way to do it in pyBind.

Reply:
Hmm, if we return the array by reference, this would mean that the array would be created by this function every time get_observation is called in python right? We're trying to store it as an attribute to avoid this.